src/inc/RangeStream.hpp

```diff
@@ -78,12 +91,15 @@ namespace MSIX {
         HRESULT STDMETHODCALLTYPE Write(const void *buffer, ULONG countBytes, ULONG *bytesWritten) noexcept override try
         {
             THROW_IF_PACK_NOT_ENABLED
-            // Forward to VectorStream/DeflateStream
-            ThrowHrIfFailed(m_stream->Write(buffer, countBytes, bytesWritten));
-            m_size = static_cast<ULONG>(m_buffer.size());
-            m_relativePosition = m_size;
-            // VectorStream will validate if the written bytes are correct. 
-            // It is expected that countBytes != bytesWritten here because this can be a compression.
+            LARGE_INTEGER offset = { 0 };
+            offset.QuadPart = m_relativePosition + m_offset;
+            ThrowHrIfFailed(m_stream->Seek(offset, StreamBase::START, nullptr));
+            ULONG amountWritten = 0;
+            ThrowHrIfFailed(m_stream->Write(buffer, countBytes, &amountWritten));
+            ThrowErrorIf(Error::FileWrite, (countBytes != amountWritten), "Did not write as much as requesteed.");

```

Comment:
>requesteed [](start = 100, length = 10)

I think you just copied my or Phil's typo...
Can you change the one in Read too?